### PR TITLE
Revamp nick command

### PR DESCRIPTION
**Pull request by Draem**
Refactor and optimization of the nick command

Also bumps lombok to 1.18.10

### CODE REVIEW

File: TGM/src/main/java/network/warzone/tgm/nickname/NickManager.java [hunk 1]
```diff
@@ -294,14 +289,10 @@ private Skin getSkin(UUID uuid) {
...
 
-    private UUID fetchUUID(String name) {
-        try {
-            HttpResponse<String> response = Unirest.get("https://api.mojang.com/users/profiles/minecraft/" + name).asString();
-            if (response.getStatus() == 200) {
-                return UUID.fromString(insertDashUUID(new JSONObject(response.getBody()).getString("id")));
-            }
-        } catch (UnirestException e) {
-            e.printStackTrace();
+    private UUID fetchUUID(String name) throws UnirestException {
```
Comment (jorgeberrex) [inline TGM/src/main/java/network/warzone/tgm/nickname/NickManager.java]: Make sure this method is called asynchronously for all its occurrences

File: TGM/src/main/java/network/warzone/tgm/nickname/NickManager.java [hunk 2]
```diff
@@ -319,19 +310,14 @@ private static String insertDashUUID(String uuid) {
...
-                JSONObject object = new JSONObject(response.getBody());
-                JSONObject properties = object.getJSONArray("properties").getJSONObject(0);
-                return new Skin(properties.getString("value"), properties.getString("signature"));
-            } else {
-                System.out.println("Connection couldn't be established code=" + response.getStatus());
-                return null;
-            }
-        } catch (UnirestException e) {
-            e.printStackTrace();
+    private Skin fetchSkin(UUID uuid) throws UnirestException {
```
Comment (jorgeberrex) [inline TGM/src/main/java/network/warzone/tgm/nickname/NickManager.java]: Run asynchronously too

File: TGM/src/main/java/network/warzone/tgm/command/MiscCommands.java [hunk 3]
```diff
@@ -14,9 +16,8 @@
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 
-import java.util.HashMap;
-import java.util.Properties;
-import java.util.UUID;
+import java.lang.reflect.InvocationTargetException;
```
Comment (jorgeberrex) [inline TGM/src/main/java/network/warzone/tgm/command/MiscCommands.java]: Unused import

File: TGM/src/main/java/network/warzone/tgm/command/MiscCommands.java [hunk 4]
```diff
@@ -49,8 +50,8 @@ public static void ping(CommandContext cmd, CommandSender sender) {
     @Command(aliases = {"nicks"}, desc = "View all nicked players")
```
Comment (jorgeberrex) [inline TGM/src/main/java/network/warzone/tgm/command/MiscCommands.java]: While this is not part of the PR, please move all the nick commands to a new dedicated class for them as the comment above the class declaration states that the commands must not implement any API functionality.

File: TGM/src/main/java/network/warzone/tgm/nickname/NickManager.java [hunk 5]
```diff
@@ -18,39 +17,42 @@
...
-import java.net.URL;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Objects;
-import java.util.UUID;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.*;
 
 public class NickManager {
```
Comment (jorgeberrex) [inline TGM/src/main/java/network/warzone/tgm/nickname/NickManager.java]: Remove unused imports

File: TGM/src/main/java/network/warzone/tgm/nickname/NickManager.java [hunk 6]
```diff
@@ -18,39 +17,42 @@
...
 
-    @Getter @Setter @AllArgsConstructor
-    public class Skin {
-        public String value;
-        public String signature;
-    }
+    public static String RATELIMITED_MESSAGE = net.md_5.bungee.api.ChatColor.GOLD.toString() + net.md_5.bungee.api.ChatColor.BOLD + "Slow Down! " + net.md_5.bungee.api.ChatColor.RESET.toString() + net.md_5.bungee.api.ChatColor.RED + "You're being ratelimited.";
```
Comment (jorgeberrex) [inline TGM/src/main/java/network/warzone/tgm/nickname/NickManager.java]: Any reason for using `net.md_5.bungee.api.ChatColor` when `org.bukkit.ChatColor` is already imported?

**Comment by jorgeberrex** [comment]
https://github.com/WarzoneMC/Warzone/blob/0a0b2471c97fd7ed9dc4f60f5b43a92ceb5eac33/TGM/src/main/java/network/warzone/tgm/command/MiscCommands.java#L209

Since this is about a revamp, perhaps it'd be beneficial to run this asynchronously

**Comment by Draem** [comment]
Resolved ✅
